### PR TITLE
Version release check

### PR DESCRIPTION
**Pull request by d0b3rm4n**
rpmvalidator just checks that rpm_release (if queryed with $RPM -q --queryformat='%{RELEASE}' -p) is equal in the file name (same for release)... but we don't check if it's 0-9_. only as we now check in Harbour upload UI... harbour-foobar-0.3.2-10.7.1.foo.armv7hl.rpm passes rpmvalidator but can't be uploaded. This adds an faulty RPM for a new test and a check into the rpmvalidator.sh.

[mersdk@SailfishSDK tests]$ ./runtest.sh
testHarbourGoodArm
testHarbourGoodArmNoColor
testHarbourBadArm
testHarbourBadArmNoColor
testHarbourGoodIntel
testHarbourBadIntel
testBadFileName
testNotAnRpm
testBadFileNameNoColor
testNotAnRpmNoColor
testHarborBadArm
testHarbourBadRpathArm
testHarbourQmlArm
testHarbourQmlNoArch
testHarbourBadRpmversionArm

Ran 15 tests.

OK
[mersdk@SailfishSDK tests]$

### CODE REVIEW

File: rpmvalidation.sh [hunk 1]
```diff
@@ -773,6 +773,18 @@ validaterpmfilename(){
     EXPECTED_RPM_FILE_NAME="${NAME}-${RPM_VERSION}-${RPM_RELEASE}.${RPM_ARCH}.rpm"
     CURRENT_RPM_FILE_NAME=$(basename $RPM_NAME)
 
+    RPM_VERSION_REGEX='^[0-9\.]+$'
+    echo $RPM_VERSION | $EGREP --silent $RPM_VERSION_REGEX
```
Comment (thp) [inline rpmvalidation.sh]: This and the next line can be rewritten as in http://stackoverflow.com/a/21112809/1047040 with something like:

```
if [[ "$RPM_VERSION" =~ "$RPM_VERSION_REGEX" ]]; then
    ...
fi
```

File: rpmvalidation.sh [hunk 2]
```diff
@@ -773,6 +773,18 @@ validaterpmfilename(){
...
+    if [[ $? -ne 0 ]] ; then
+        validation_error $RPM_VERSION "rpm version must contain only digits (0-9) and periods (.)"
+    fi
+
+    RPM_RELEASE_REGEX='^[0-9_\.]+$'
+    echo $RPM_RELEASE | $EGREP --silent $RPM_RELEASE_REGEX
```
Comment (thp) [inline rpmvalidation.sh]: Same as above

File: tests/runtest.sh [hunk 3]
```diff
@@ -162,6 +162,12 @@ function testHarbourQmlNoArch() {
     assertTrue 'Expected output differs.' $?
 }
 
+function testHarbourBadRpmversionArm() {
+    ${RPMVALITATOR} ${RPMS_DIR}/harbour-bad-rpmversion-0.1a-1.jolla.armv7hl.rpm 2>&1 | ${SED_FILTER} > ${OUTPUT_DIR}/testHarbourBadRpmversionArm.txt
```
Comment (thp) [inline tests/runtest.sh]: Typo `RPMVALITATOR` → `RPMVALIDATOR`, but probably that typo fix should be done in a separate PR.

**Comment by thp** [comment]
Please use `=~` from bash as documented in http://www.gnu.org/software/bash/manual/bashref.html#Conditional-Constructs for matching a string against a regex.

**Comment by d0b3rm4n** [comment]
yeah due to different devels we have a mix of bashism and pure posix shell :) but we can start using =~ I will update it... the RPMVALITATOR is wrong in the whole runtests.sh script.

**Comment by thp** [comment]
Yup. We already use `bash` in the shebang line, so we should make use of its features (at some point "when there is time", I'd like to see the script at least partially rewritten in a non-shell scripting language, let's see when we have time for that ;).

**Comment by thp** [comment]
LGTM, thanks :)
